File: RefactorMe/C#/RefactorMe/ProductDataConsolidator.cs
using RefactorMe.DontRefactor.Data.Implementation;
using RefactorMe.DontRefactor.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RefactorMe
{
    public class ProductDataConsolidator
    {
        public static List<Product> Get() {
            var l = new LawnmowerRepository().GetAll();
            var p = new PhoneCaseRepository().GetAll();
            var t = new TShirtRepository().GetAll();

            var ps = new List<Product>();

            foreach (var i in l) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price,
                    Type = "Lawnmower"
                });
            }

            foreach (var i in p) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price,
                    Type = "Phone Case"
                });
            }

            foreach (var i in t) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price,
                    Type = "T-Shirt"
                });
            }

            return ps;
        }

        public static List<Product> GetInUSDollars() {
            var l = new LawnmowerRepository().GetAll();
            var p = new PhoneCaseRepository().GetAll();
            var t = new TShirtRepository().GetAll();

            var ps = new List<Product>();

            foreach (var i in l) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price * 0.76,
                    Type = "Lawnmower"
                });
            }

            foreach (var i in p) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price * 0.76,
                    Type = "Phone Case"
                });
            }

            foreach (var i in t) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price * 0.76,
                    Type = "T-Shirt"
                });
            }

            return ps;
        }

        public static List<Product> GetInEuros() {
            var l = new LawnmowerRepository().GetAll();
            var p = new PhoneCaseRepository().GetAll();
            var t = new TShirtRepository().GetAll();

            var ps = new List<Product>();

            foreach (var i in l) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price * 0.67,
                    Type = "Lawnmower"
                });
            }

            foreach (var i in p) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price * 0.67,
                    Type = "Phone Case"
                });
            }

            foreach (var i in t) {
                ps.Add(new Product() {
                    Id = i.Id,
                    Name = i.Name,
                    Price = i.Price * 0.67,
                    Type = "T-Shirt"
                });
            }

            return ps;
        }
 
 
    }
}

File: workingout/JavaScript/src/refactor-me/ProductDataConsolidator.js
function ProductDataConsolidator() { }

ProductDataConsolidator.get = function () {
	var l = new LawnmowerRepository().getAll();
	var p = new PhoneCaseRepository().getAll();
	var t = new TShirtRepository().getAll();

	var products = [];

	for (var i = 0; i < l.length; i++) {
		products.push({
			id: l[i].id,
			name: l[i].name,
			price: l[i].price.toFixed(2),
			type: "Lawnmower"
		});
	}

	for (var i = 0; i < p.length; i++) {
		products.push({
			id: p[i].id,
			name: p[i].name,
			price: p[i].price.toFixed(2),
			type: "Phone Case"
		});
	}

	for (var i = 0; i < t.length; i++) {
		products.push({
			id: t[i].id,
			name: t[i].name,
			price: t[i].price.toFixed(2),
			type: "T-Shirt"
		});
	}

	return products;
}

ProductDataConsolidator.getInUSDollars = function () {
	var l = new LawnmowerRepository().getAll();
	var p = new PhoneCaseRepository().getAll();
	var t = new TShirtRepository().getAll();

	var products = [];

	for (var i = 0; i < l.length; i++) {
		products.push({
			id: l[i].id,
			name: l[i].name,
			price: (l[i].price * 0.76).toFixed(2),
			type: "Lawnmower"
		});
	}

	for (var i = 0; i < p.length; i++) {
		products.push({
			id: p[i].id,
			name: p[i].name,
			price: (p[i].price * 0.76).toFixed(2),
			type: "Phone Case"
		});
	}

	for (var i = 0; i < t.length; i++) {
		products.push({
			id: t[i].id,
			name: t[i].name,
			price: (t[i].price * 0.76).toFixed(2),
			type: "T-Shirt"
		});
	}

	return products;
}

ProductDataConsolidator.getInEuros = function () {
	var l = new LawnmowerRepository().getAll();
	var p = new PhoneCaseRepository().getAll();
	var t = new TShirtRepository().getAll();

	var products = [];

	for (var i = 0; i < l.length; i++) {
		products.push({
			id: l[i].id,
			name: l[i].name,
			price: (l[i].price * 0.67).toFixed(2),
			type: "Lawnmower"
		});
	}

	for (var i = 0; i < p.length; i++) {
		products.push({
			id: p[i].id,
			name: p[i].name,
			price: (p[i].price * 0.67).toFixed(2),
			type: "Phone Case"
		});
	}

	for (var i = 0; i < t.length; i++) {
		products.push({
			id: t[i].id,
			name: t[i].name,
			price: (t[i].price * 0.67).toFixed(2),
			type: "T-Shirt"
		});
	}

	return products;
}
